### DISCUSSION TITLE
How to run the Oracle SOA Suite Docker with a Different Fully Privileged User other than the sys user?

### DISCUSSION THREAD

**Discussion by ahfarag** [oracle/docker-images#2846] [Q&A]
I'm in the process of setting up Oracle SOA Suite using Docker documented [here](https://github.com/oracle/docker-images/tree/main/OracleSOASuite/dockerfiles/12.2.1.4), and I've encountered a configuration issue. The container setup requires a parameter `DB_PASSWORD` for the `sys` user password of the Oracle database as mentioned in this section [5. Create a container for the Administration Server](https://github.com/oracle/docker-images/tree/main/OracleSOASuite/dockerfiles/12.2.1.4#5-create-a-container-for-the-administration-server). However, I have another database user with all the necessary privileges and would prefer to use this user instead of the SYS user.

Does anyone know how I can configure or run the Oracle SOA Suite Docker container to use this alternative privileged user? Are there specific environment variables or configuration files that I need to adjust?

Thanks in advance for your help!

**Comment by Bulletdev** [comment]
To configure the Oracle SOA Suite Docker container to use an alternative privileged user instead of the SYS user, you can follow these steps:

Steps to Use an Alternative Privileged User
1.
Create a Custom Docker Image:
• Start by creating a custom Docker image based on the official Oracle SOA Suite Docker image.

• In your Dockerfile, you can modify the entry point or initialization scripts to use your alternative user.

Modify Environment Variables:
• You can set environment variables to use your alternative user. For example, you might need to set DB_USER and DB_PASSWORD to your alternative user's credentials.
2.
Update Configuration Files:
• If there are specific configuration files that reference the SYS user, update them to use your alternative user. This might include files like datasource.xml or other database connection configuration files.

Example Dockerfile
Here is an example of how you might modify your Dockerfile:

FROM oracle/soasuite:12.2.1.4
Set environment variables for the alternative user

ENV DB_USER=my_alternative_user
ENV DB_PASSWORD=my_alternative_password
Copy custom scripts or configuration files if needed

COPY custom-init.sh /u01/oracle/
Run the custom initialization script

RUN chmod +x /u01/oracle/custom-init.sh
ENTRYPOINT ["/u01/oracle/custom-init.sh"]

Example Custom Initialization Script
Here is an example of a custom initialization script (custom-init.sh):

#!/bin/bash
Custom initialization logic
Replace SYS user references with the alternative user
Example: Update datasource configuration

sed -i 's/SYS_USER/${DB_USER}/g' /u01/oracle/user_projects/domains/soa_domain/config/jdbc/datasource.xml
sed -i 's/SYS_PASSWORD/${DB_PASSWORD}/g' /u01/oracle/user_projects/domains/soa_domain/config/jdbc/datasource.xml
Start the Oracle SOA Suite server

/u01/oracle/startWebLogic.sh

Additional Resources
• Oracle Documentation: Refer to the official Oracle documentation for any specific environment variables or configuration settings required for your setup.

• Community Support: Check forums and community discussions for any additional tips or shared experiences from other users.

By following these steps, you should be able to configure the Oracle SOA Suite Docker container to use your alternative privileged user. If you encounter any issues or need further assistance, feel free to ask! https://support.oracle.com/knowledge/Middleware/3046883_1.html Oracle SOA Suite Docker Configuration #2846 Oracle Support Document on Using Alternative User
